### ISSUE TITLE
Consider changing default Mypy Arguments setting to avoid spurious errors when used with warn_unused_ignores

### ISSUE THREAD

**Issue by PeterJCLaw** [microsoft/vscode-mypy#192]
## Environment data

- VS Code version: 1.46.1
- Extension version (available under the Extensions sidebar): 2020.6.90262

## Expected behaviour

I generally expect that my editor will give the same linter experience as running a linter directly on the command line.

## Actual behaviour

For the case of `warn_unused_ignores` in combination with a `type: ignore` comment on an import line, running `mypy` through VSCode differs from the command line experience and thus from what any scripted runs (including CI) would output. This results from the vscode-python extension passing non-default arguments to `mypy` which override behaviours the user may have configured in their project.

## Steps to reproduce:

``` ini
# setup.cfg
[mypy]
warn_unused_ignores = True
```

``` python
# demo.py
import asttokens  # type: ignore

print(asttokens)
```

<details>

``` shell
pip install asttokens==2.0.4 mypy
```

I'm using `asttokens` here, but I've seen this with other libraries too (for example `parso` and `jedi`). The common theme is that it is a library which doesn't have type annotations.

Command line run:
```
$ mypy demo.py 
demo.py:9: error: Function is missing a type annotation
Success: no issues found in 1 source file
```

VSCode Output tab output:
```
demo.py:1: error: unused 'type: ignore' comment
Found 1 error in 1 file (checked 1 source file)
```

</details>

---

I had initially assumed that this was a bug in vscode-python and it was only while reviewing the `mypy` setting to check I didn't have any changes I should include in the bug report that I spotted the non-default arguments which vscode-python passes to `mypy`.

I can definitely see why the chosen options were set -- to reduce the noise from `mypy` for new users who may not have configured `mypy` themselves, however this leads to a non-obvious difference between `mypy` in the editor and `mypy` on the command line.

I can see that it may be somewhat reasonable for an editor to have false negatives (or hide errors) relative to the command line, however the issue here is that these options create false _positives_ -- errors in the editor which it is not clear how to resolve and which do not reproduce elsewhere. This leads to a reduction in trust that the editor and/or `mypy` is reporting the right things.
